traitsui/tests/_tools.py

```diff
@@ -131,6 +131,30 @@ def filter_tests(test_suite, exclusion_pattern):
     return filtered_test_suite
 
 
+@contextmanager
+def create_ui(object, ui_kwargs=None):
+    """ Context manager for creating a UI and then dispose it when exiting
+    the context.
+
+    Parameters
+    ----------
+    object : HasTraits
+        An object from which ``edit_traits`` can be called to create a UI
+    ui_kwargs : dict or None
+        Keyword arguments to be provided to ``edit_traits``.
+
+    Yields
+    ------
+    ui: UI
+    """
+    ui_kwargs = {} if ui_kwargs is None else ui_kwargs
+    ui = object.edit_traits(**ui_kwargs)
```

Comment:
There are already a few context manager that do the same + yield an editor (in `test_date_editor.py`, `test_date_range_editor.py`, `test_datetime_editor.py`). I would say that working with an editor is a common enough use case to yield it here together with `ui`, potentially with an option to specify editor by "name" (e.g. "txt" as in tests below). And then the other context managers can be replaced with this one.

Reply:
Agreed that it is a common pattern to try to retrieve an editor with a name.

On the other hand, getting an editor may fail, mixing it with launching the GUI makes the traceback harder to understand. Sometimes one may also just launch the GUI and close it. The signature of the context manager will then have to allow `None` for the maybe-not-needed editor... `editor, = ui.get_editors("name")` is easy enough to me.  
See also https://github.com/enthought/traitsui/pull/861 for an even easier API for the most common use cases, where you don't need to call `get_editors` at all. :)